waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java

```diff
@@ -77,14 +83,28 @@ public DatabaseMappingImpl(MetaStoreMapping metaStoreMapping, QueryMapping query
   public Table transformOutboundTable(Table table) {
     table.setDbName(metaStoreMapping.transformOutboundDatabaseName(table.getDbName()));
     if (table.isSetViewExpandedText()) {
-      table.setViewExpandedText(
-          queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewExpandedText()));
+      try {
+        log.debug("Transforming ViewExpandedText: {}", table.getViewExpandedText());
+        table.setViewExpandedText(
+            queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewExpandedText()));
+      } catch (WaggleDanceException e) {
+        log.debug("Error while transforming databaseName in ViewExpandedText, keeping original", e);
+      }
     }
     if (table.isSetViewOriginalText()) {
-      table.setViewOriginalText(
-          queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewOriginalText()));
+      try {
+        log.debug("Transforming ViewOriginalText: {}", table.getViewOriginalText());
+        table.setViewOriginalText(
+            queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewOriginalText()));
+      } catch (WaggleDanceException e) {
+        log.debug("Error while transforming databaseName in ViewOriginalText, using ViewExpandedText if available", e);
+        if (table.isSetViewExpandedText()) {
+          table.setViewOriginalText(table.getViewExpandedText());
```

Comment:
not sure about this, is this a temporary fix? can you put a comment explaining why this is done here?

Reply:
The fix is permanent because we can always hit the case where the QueryMapping fails and we shouldn't fail the client. But I can add a comment with the link to the Hive bug (https://issues.apache.org/jira/browse/HIVE-19896)